src/common/io.ts

```diff
@@ -154,38 +154,48 @@ export class ARIOReadable implements AoARIORead {
     });
   }
 
+  private async computeEpochIndexForTimestamp(
+    timestamp: number,
+  ): Promise<number> {
+    const epochSettings = await this.getEpochSettings();
+    const epochZeroStartTimestamp = epochSettings.epochZeroStartTimestamp;
+    const epochLengthMs = epochSettings.durationMs;
+    return Math.floor((timestamp - epochZeroStartTimestamp) / epochLengthMs);
+  }
+
+  private async epochInputToEpochIndexTag(
+    params?: EpochInput,
+  ): Promise<{ name: string; value: string }> {
+    const epochIndex = (params as { epochIndex?: number })?.epochIndex;
+
+    if (epochIndex !== undefined) {
+      return { name: 'Epoch-Index', value: epochIndex.toString() };
+    }
+    const timestamp = (params as { timestamp?: number })?.timestamp;
+    if (timestamp !== undefined) {
+      return {
+        name: 'Epoch-Index',
+        value: (await this.computeEpochIndexForTimestamp(timestamp)).toString(),
+      };
+    }
+    return { name: 'Epoch-Index', value: '' };
+  }
```

Comment:
```suggestion
  private async computeEpochIndex(
    params?: EpochInput,
  ): Promise<number | undefined> {
   const { epochIndex = undefined, timestamp = undefined } = params; // you can just ts-ignore
    if (timestamp){
        epochIndex ?= await await this.computeEpochIndexForTimestamp(timestamp)).toString()
    }
    return epochIndex
```
and then keep the `Epoch-Index` tag setup in the caller

Reply:
[42399a4](https://github.com/ar-io/ar-io-sdk/pull/325/commits/42399a42acc96b796edff30d03824034ea1c3e58)